### PR TITLE
WIP: [PLT-7364] Ability to Direct Message yourself

### PR DESCRIPTION
**Pull request by cpanato**
I need some help to test and find if this is good in terms of code.
@jasonblais can help me here?

Server PR: https://github.com/mattermost/mattermost-server/pull/7696
Webapp PR: https://github.com/mattermost/mattermost-webapp/pull/186

#### Summary
 Ability to Direct Message yourself

#### Ticket Link
Jira: https://mattermost.atlassian.net/browse/PLT-7364
GitHub: https://github.com/mattermost/mattermost-server/issues/7618

### CODE REVIEW

File: src/actions/users.js [hunk 1]
```diff
@@ -241,7 +241,9 @@ export function getProfiles(page = 0, perPage = General.PROFILE_CHUNK_SIZE) {
         let profiles;
         try {
             profiles = await Client4.getProfiles(page, perPage);
-            removeUserFromList(currentUserId, profiles);
+            if (skipCurrent) {
```
Comment (enahum) [inline src/actions/users.js]: why is this conditional?

The problem by not skipping the current user all the time is that you will replace the currentUser with whatever getProfiles returns and then cause of the profile being sanitize you'll remove the `notifiy_props` needed by the currentUser

File: src/actions/users.js [hunk 2]
```diff
@@ -241,7 +241,9 @@ export function getProfiles(page = 0, perPage = General.PROFILE_CHUNK_SIZE) {
...
+                removeUserFromList(currentUserId, profiles);
+            }
```
Comment (jwilander) [inline src/actions/users.js]: Looking at your webapp code, is this change really needed? The redux store should already have the current user and you're not using the direct action result anywhere you're just selectors to pull from the store, which is guaranteed to already have the current user.

Let me know if I'm missing something

Comment (cpanato) [reply src/actions/users.js]: tested, dont need this change here, closing...

File: src/actions/users.js (hunk 1 above)
Comment (cpanato) [reply src/actions/users.js]: tested, dont need this change here, closing...

**Comment by cpanato** [comment]
tested, dont need this change here, closing...
